concordium-std/src/impls.rs

```diff
@@ -762,13 +813,14 @@ where
         key_with_prefix
     }
 
-    pub fn iter(&self) -> StateSetIter<T, S> {
-        let state_iter = self.state_ll.borrow().iterator(&self.prefix);
-        StateSetIter {
+    pub fn iter(&self) -> Result<StateSetIter<T, S>, ContractStateError> {
```

Comment:
Same idea as with the iter for statemap. It should have a reference to the stateset.

Reply:
Yes.

I would just like you don't get bogged down in this part and that we get something working if the references and lifetimes are not the most straightforward concept.

We can do it in stages.